bin.src/makeLSSTobs.py

```diff
@@ -13,21 +13,49 @@
 from lsst.sims.maf.batches import getColMap
 
 
-def readOpsim(opsimfile, constraint=None, dbcols=None):
+def readOpsim(opsimfile, constraint=None, footprint='camera', dbcols=None):
     # Read opsim database.
     opsdb = OpsimDatabase(opsimfile)
-    if dbcols is None:
-        dbcols = []
+
     colmap = getColMap(opsdb)
     if 'rotSkyPos' not in colmap:
         colmap['rotSkyPos'] = 'rotSkyPos'
-    reqcols = [colmap['mjd'], colmap['night'], colmap['ra'], colmap['dec'],
-               colmap['rotSkyPos'], colmap['filter'], colmap['exptime'], colmap['seeingEff'],
-               colmap['seeingGeom'], colmap['fiveSigmaDepth'], 'solarElong']
-    for col in reqcols:
-        if col not in dbcols:
-            dbcols.append(col)
-    simdata = opsdb.fetchMetricData(dbcols, sqlconstraint=constraint)
+
+    # Set the minimum required columns.
+    min_cols = [colmap['mjd'], colmap['night'], colmap['ra'], colmap['dec'],
+                colmap['filter'], colmap['exptime'], colmap['seeingGeom'],
+                colmap['fiveSigmaDepth']]
+    if footprint == 'camera':
+        min_cols.append(colmap['rotSkyPos'])
+    if dbcols is not None:
+        min_cols += dbcols
+    min_cols = list(set(min_cols))
+
+    # Check if these minimum required columns are in the database.
+    simdata = opsdb.query_columns(tablename=opsdb.defaultTable,
+                                  colnames=min_cols,
+                                  sqlconstraint=constraint,
+                                  numLimit=1)
+
+    # If that was successful, there are some additional columns that can be useful:
+    more_cols = [colmap['rotSkyPos'], colmap['seeingEff'], 'solarElong']
+    failed_cols = []
+    for col in more_cols:
+        try:
+            simdata = opsdb.query_columns(tablename=opsdb.defaultTable,
+                                          colnames=[col], sqlconstraint=constraint, numLimit=1)
+        except ValueError:
+            failed_cols.append(col)
+    if len(failed_cols) > 0:
+        print('Could not find columns %s in the database, so leaving those out for now.' % failed_cols)
```

Comment:
If the message comes up, the user doesn't really have to *deal* with it, but they might like to know about it (generally because it's likely to be either unexpected to them, or it's expected and then they can ignore it). I guess I'm assuming the user kind of knows what's in their database and if something unexpected shows up here (couldn't find some column that they thought was there), they might want to check into it a bit more.

I haven't used proper logging .. you're probably right that this is what should be happening, if only for better redirect capability. 

Actual problematic errors anywhere in the code will raise exceptions and halt execution. Anything else is just INFO level.  To me, DEBUG seems like you wouldn't have it on in usual execution?

Reply:
I'm torn between making this an INFO (with a default logging level of WARN) or WARN (but raise the default logging level to ERROR). Either way, I think the users shouldn't have it displayed by default -- if they encounter an error they could increase the log level and get this extra information.

Why: My guess is the vast majority of users won't know what this message is about -- they'll be using the code as a black box, with databases we've given them (with maybe an ε more who have read the documentation :-)). A good rule of thumb may be to write messages to the console if and only if the user should act on them. Otherwise they get confused and either don't trust the result ("oh my, look at all these messages, this can't be good!"), or stop looking at the messages all together and fail to notice the really important ones that indicate something /is/ fishy with the result.

Since Python's default logging level is WARN, that would make most of these types of messages INFO. Or one could configure the logger to raise the console output level to ERROR or CRITICAL by default, so these types of things could be WARN.

Here are [Python's official guidelines](https://docs.python.org/3/howto/logging.html#when-to-use-logging). LSST pipelines [by default log INFO to the console](http://doxygen.lsst.codes/stack/doxygen/x_masterDoxyDoc/log.html), though they're not intended for broad use (and generally dump way too much output to the console). 

PS: Or maybe even DEBUG -- if we follow the Python guidelines, these messages will only be used when diagnosing problems, and that's what DEBUG is for (INFO is for "confirmation that things are working as expected" whereas DEBUG is for "Detailed information, typically of interest only when diagnosing problems.")

Reply:
I guess I view these as "Info" - "confirmation that things are working as expected".
I did set it so that info comments were sent to the console .. I guess I feel there's not that many of them and they are helpful for users to know what's going on when.

Reply:
I also updated any 'print' statements to logging messages.